tests/johnsnowlabs/test_jsl_api.py

```diff
@@ -0,0 +1,782 @@
+from tests.helper_functions import (
+    _compare_conda_env_requirements,
+    _assert_pip_requirements,
+    _compare_logged_code_paths,
+    _mlflow_major_version_string, )
+from tests.helper_functions import score_model_in_sagemaker_docker_container
+
+# TODO you must update this
+license_keys = {
+    "AWS_ACCESS_KEY_ID": None,
+    "AWS_SECRET_ACCESS_KEY": None,
+    "SPARK_NLP_LICENSE": None,
+    "SECRET": None}
+
+os.environ.update(license_keys)
+
+# Before running any tests make sure the following environment variables are set:
+
+if not license_keys['SECRET']:
+    raise ValueError("""
+Please set SECRET, AWS_ACCESS_KEY_ID ,AWS_SECRET_ACCESS_KEY ,SPARK_NLP_LICENSE in your environment variables
+""")
+
+# nlu_model = 'tokenize' # You can use this alternatively for a leightweight test run with a tokenizer model
+nlu_model = 'en.classify.bert_sequence.covid_sentiment'
+
+
+def load_and_init_model(model=nlu_model):
+    nlp.start(model_cache_folder='/home/ckl/dump/cache_pretrained')
+    jsl_model = nlp.load(model, verbose=False)
+    return jsl_model
+
+
+def fix_dataframe_with_respect_for_nlu_issues(d1, d2):
+    # TODO there may be some changes in confidence and changes in column names after storing/loading a model
+    # these issues in NLU which are not related to MLflow and to be fixed.
+    # For now we are applying a hotfix here on the dataframes to make sure that the tests run the way they should
+    d1 = d1.drop(columns=[c for c in d1.columns if c not in d2.columns or 'confidence' in c])
+    d2 = d2.drop(columns=[c for c in d2.columns if c not in d1.columns or 'confidence' in c])
+
+    def lower_strings(df):
+        for c in df.columns:
+            try:
+                df[c] = df[c].str.lower()
+            except:
+                pass
+        return df
+
+    d1 = lower_strings(d1)
+    d2 = lower_strings(d2)
+    return d1, d2
+
+
+def validate_model(original_model, new_model):
+    d1, d2 = original_model.predict("Hello World"), new_model.predict("Hello World")
+    if isinstance(d2, str):
+        d2 = pd.DataFrame(json.loads(d2)).drop(columns=['index']).reset_index().drop(columns=['index'])
+    else:
+        d2 = d2.reset_index().drop(columns=['index'])
+    d1 = d1.reset_index().drop(columns=['index'])
+
+    # TODO fix: column names may change before/after save and Confidences change
+    d1 = d1.drop(columns=[c for c in d1.columns if 'confidence' in c])
+    d2 = d2.drop(columns=[c for c in d2.columns if 'confidence' in c])
+    d1.columns = [f'c_{i}' for i in range(len(d1.columns))]
+    d2.columns = [f'c_{i}' for i in range(len(d2.columns))]
+    assert d1.equals(d2)
+
+
+@pytest.fixture
+def jsl_model(model_path):
+    model = load_and_init_model()
+    yield model
+    # shutil.rmtree(model_path)
+
+
+@pytest.fixture
+def model_path(tmpdir):
+    return os.path.join(str(tmpdir), "model")
+
+
+@pytest.fixture
+def spark_custom_env(tmpdir):
+    conda_env = os.path.join(str(tmpdir), "conda_env.yml")
```

Comment:
```suggestion
    conda_env = tmpdir.joinpath("conda_env.yml")
```

Reply:
This gives ` AttributeError: 'LocalPath' object has no attribute 'joinpath'`  using`Path(str(tmpdir)) / "conda_env.yml"` instead